components/infobox/wikis/pubg/infobox_league_custom.lua

```diff
@@ -0,0 +1,224 @@
+		end
+
+		--teams section
+		if _args.team_number then
+			table.insert(widgets, Title{name = 'Teams'})
+			table.insert(widgets, Cell{name = 'Number of teams', content = {_args.team_number}})
+		end
+	end
+	return widgets
+end
+
+function CustomLeague:addToLpdb(lpdbData, args)
+	lpdbData.game = _game or args.game
+	lpdbData.participantsnumber = args.player_number or args.team_number
+	lpdbData.extradata = {
+		individual = String.isNotEmpty(args.player_number) and 'true' or '',
+	}
+
+	return lpdbData
+end
+
+function CustomLeague:defineCustomPageVariables()
+	Variables.varDefine('tournament_game', _game or _args.game)
+	Variables.varDefine('tournament_publishertier', _args['pubgpremier'])
+	--Legacy Vars:
+	Variables.varDefine('tournament_edate', Variables.varDefault('tournament_enddate'))
+end
+
+function CustomLeague._getGameVersion()
+	local game = string.lower(_args.game or '')
+	_game = _GAME[game]
+	return _game
+end
+
+function CustomLeague:_createPrizepool()
+	if String.isEmpty(_args.prizepool) and String.isEmpty(_args.prizepoolusd) then
+		return nil
+	end
+	local date
+	if String.isNotEmpty(_args.currency_rate) then
+		date = _args.currency_date
+	end
+
+	return PrizePoolCurrency._get({
+		prizepool = _args.prizepool,
+		prizepoolusd = _args.prizepoolusd,
+		currency = _args.localcurrency,
+		rate = _args.currency_rate,
+		date = date or Variables.varDefault('tournament_enddate', _TODAY),
+	})
+end
+
+function CustomLeague:_createTierDisplay()
+	local tier = _args.liquipediatier or ''
+	local tierType = _args.liquipediatiertype or _args.tiertype or ''
+	if String.isEmpty(tier) then
+		return nil
+	end
+
+	local tierText = Tier['text'][tier]
+	local hasInvalidTier = tierText == nil
+	tierText = tierText or tier
+
+	local hasInvalidTierType = false
+
+	local output = '[[' .. tierText .. ' Tournaments|' .. tierText .. ']]'
+		.. '[[Category:' .. tierText .. ' Tournaments]]'
+
+	if not String.isEmpty(tierType) then
+		tierType = Tier['types'][string.lower(tierType or '')] or tierType
+		hasInvalidTierType = Tier['types'][string.lower(tierType or '')] == nil
+		tierType = '[[' .. tierType .. ' Tournaments|' .. tierType .. ']]'
+			.. '[[Category:' .. tierType .. ' Tournaments]]'
+		output = tierType .. '&nbsp;(' .. output .. ')'
+	end
+
+	output = output ..
+		(hasInvalidTier and '[[Category:Pages with invalid Tier]]' or '') ..
+		(hasInvalidTierType and '[[Category:Pages with invalid Tiertype]]' or '')
+	return output
+end
+
+function CustomLeague:_getGameMode()
+	if String.isEmpty(_args.perspective) and String.isEmpty(_args.mode) then
+		return nil
+	end
+
+	local perspective = string.lower(_args.perspective or '')
+	perspective = string.gsub(perspective, ' person', '')
+	perspective = string.gsub(perspective, ' perspective', '')
+	perspective = _PERSPECTIVES[perspective] or {}
+	for key, item in pairs(perspective) do
+		perspective[key] = Template.safeExpand(mw.getCurrentFrame(), 'Abbr/' .. item)
+	end
```

Comment:
```suggestion
	local getPerspectives = function(perspectiveInput)
		local perspectives = string.lower(perspectiveInput or '')
		-- Clean unnecessary data from the input
		perspective = string.gsub(perspective, ' person', '')
		perspective = string.gsub(perspective, ' perspective', '')
		return _PERSPECTIVES[perspective] or {}
	end
	local getPerspectiveDisplay = function(perspective) 
		return Template.safeExpand(mw.getCurrentFrame(), 'Abbr/' .. perspective) 
	end
	
	local displayPerspectives = Table.mapValues(getPerspectives(_args.perspective), getPerspectiveDisplay) 
```
And update the table.concat a few lines down to use displayPerspectives

Reply:
Got **212: attempt to index global 'Table' (a nil value).**

Pics include the concat a few lines after
![image](https://user-images.githubusercontent.com/88981446/152982073-e9ddd651-8e09-47c6-85e2-85f6cb07af30.png)

Reply:
add `local Table = require('Module:Table')` at the top

Reply:
> add `local Table = require('Module:Table')` at the top

Ok, 212 has been solved but has return another error at 205 mentioned something with **gsub**
![image](https://user-images.githubusercontent.com/88981446/152995750-54208939-8e04-4d76-9806-6fee2b3de343.png)

Reply:
change `local perspectives = string.lower(perspectiveInput or '')` to `local perspective = string.lower(perspectiveInput or '')`
(the `perspective` on line 205 is nil because it never gets set because in line 203 it is `perspectives`, with the `s` at the end)